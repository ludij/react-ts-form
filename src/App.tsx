import React from "react"
import CForm from "./components/CForm"

const formFields = [
  {
    type: "text",
    label: "First Name",
    name: "first-name",
    value: "",
  },
  {
    type: "text",
    label: "Last Name",
    name: "last-name",
    value: "",
  },
  {
    type: "radio",
    label: "Income",
    name: "income",
    options: [
      {
        label: "Option 1",
        value: "option-1",
        checked: false,
      },
      {
        label: "Option 2",
        value: "option-2",
        checked: false,
      },
      {
        label: "Option 3",
        value: "option-3",
        checked: true,
      },
    ],
  },
]

function App() {
  return (
    <div>
      <CForm formFields={formFields} />
    </div>
  )
}

export default App
