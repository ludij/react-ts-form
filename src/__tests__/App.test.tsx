import React from "react"
import renderer from "react-test-renderer"
import { render, screen } from "@testing-library/react"
import serializer from "jest-emotion"
import App from "../App"

expect.addSnapshotSerializer(serializer)

test("renders learn react link", () => {
  const component = render(<App />)
  const test = component.getByText("test")
  expect(screen.getByText("test")).toBe(test)
})

test("renderer link", () => {
  const component = renderer.create(<App />).toJSON()
  expect(
    component.children.filter((item : any) => item.type === "button").length === 1
  )
  expect(component).toMatchSnapshot()
})
