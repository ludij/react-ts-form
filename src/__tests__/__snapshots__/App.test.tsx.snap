// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer link 1`] = `
.emotion-0 {
  color: red;
}

<div
  className="App"
>
  <button
    className="emotion-0 emotion-1"
  >
    test
  </button>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
